### PR TITLE
Tensor product cleanup

### PR DESCRIPTION
**Pull request by HechtiDerLachs**
There was a silent assumtion on the `preimage` functionality for projection maps made in the fix for tensor products. This hopefully "generifies" the code again and fixes up the induced Eagon-Northcott complexes.

Moreover, I straighten out some issues with the "homogeneous Koszul complexes" after the recent switch to `Combinations`.

### CODE REVIEW

**Comment by lgoettgens** [comment]
Who can review this? @oscar-system/algebraic-geometry

**Comment by HechtiDerLachs** [comment]
Maybe @jankoboehm can have a quick look? We talked about this issue yesterday and he fixed it in his PR on modules over the integers by different means. 

The part about the induced Eagon-Northcott complexes is independent of that issue, but I vouch for that part on my end.

File: src/Combinatorics/EnumerativeCombinatorics/combinations.jl [hunk 1]
```diff
@@ -150,7 +150,7 @@ julia> Oscar.linear_index(C, 7)
 ```
 """
 function linear_index(C::Combination, n::IntegerUnion)
-  @req C[end] <= n "Combination must be bounded by n"
+  @req is_empty(C) || C[end] <= n "Combination must be bounded by n"
```
Comment (HechtiDerLachs) [inline src/Combinatorics/EnumerativeCombinatorics/combinations.jl:153]: @mjrodgers : It seems like there was an edge case which was not covered: The linear index of the empty list (which is always one). I hope you can approve of this change.

Comment (mjrodgers) [reply src/Combinatorics/EnumerativeCombinatorics/combinations.jl:153]: Absolutely, I should have had a test for an edge case like this...

**Comment by jankoboehm** [comment]
> Maybe @jankoboehm can have a quick look? We talked about this issue yesterday and he fixed it in his PR on modules over the integers by different means.
> 
> The part about the induced Eagon-Northcott complexes is independent of that issue, but I vouch for that part on my end.

The fix in the other PR is indeed also generic and gives a general performance improvement. Nevertheless the fix here is useful if someone uses the machinery bypassing the abstraction layer.

**Review by jankoboehm — APPROVED** [review]
